pkg/rtc/participant.go

```diff
@@ -466,8 +466,16 @@ func (p *ParticipantImpl) SetAttributes(attrs map[string]string) error {
 	if grants.Attributes == nil {
 		grants.Attributes = make(map[string]string)
 	}
+	var keysToDelete []string
 	for k, v := range attrs {
-		grants.Attributes[k] = v
+		if v == "" {
+			keysToDelete = append(keysToDelete, k)
+		} else {
+			grants.Attributes[k] = v
+		}
+	}
+	for _, k := range keysToDelete {
+		delete(grants.Attributes, k)
 	}
```

Comment:
nit does copilot write map deletes this way? i feel like i saw this in a different diff recently...

Reply:
lol.. just me dude. I've always written it this way. do you have a suggested alternative?

Reply:
`keysToDelete = append(keysToDelete, k)` => `delete(grants.Attributes, k)`. is this just force of habit or is there an advantage to doing it this way?

Reply:
just habit of not modifying the map during iteration. probably a habit from the java days :)